vms/platformvm/state/stakers.go

```diff
@@ -27,6 +35,12 @@ type CurrentStakers interface {
 	// Invariant: [staker] is not currently a CurrentValidator
 	PutCurrentValidator(staker *Staker)
 
+	// UpdateCurrentValidator updates the [staker] describing a validator to the
+	// staker set.
+	//
+	// Invariant: [staker] is a non deleted CurrentValidator
+	UpdateCurrentValidator(staker *Staker) error
```

Comment:
note: we don't need updating pending validators (we plan to remove pending stakers completelly with continuous staking)